.changeset/flat-snakes-hammer.md

```diff
@@ -0,0 +1,50 @@
+---
+"@astrojs/vercel": minor
+---
+
+Introduces a new config options, `isr`, that allows to deploy your project as an ISR function. [ISR (Incremental Static Regeneration)](https://vercel.com/docs/incremental-static-regeneration) caches your on-demand rendered pages in the same way as prerendered pages after first request.
+
+To enable this feature, set `isr` to true in your `astro.config.mjs`:
+
+```js
+export default defineConfig({
+    output: "server",
+    adapter: vercel({ isr: true })
+})
+```
+
+##### Time-based invalidation
+
+By default, ISR responses are cached for the duration of your deployment. You can change this by setting `isr` to an object with an `expiration` field:
+
+```js
+export default defineConfig({
+    output: "server",
+    adapter: vercel({
+        isr: {
+            // caches all pages on first request and saves for 1 day
+            expiration: 60 * 60 * 24
+        }
+    })
+})
+```
+
+##### Manual invalidation
+
+To implement [Draft mode](https://vercel.com/docs/build-output-api/v3/features#draft-mode), or [On-Demand Incremental Static Regeneration (ISR)](https://vercel.com/docs/build-output-api/v3/features#on-demand-incremental-static-regeneration-isr), you can create a bypass token and provide it to the `isr` config: 
+
+```js
+export default defineConfig({
+    output: "server",
+    adapter: vercel({
+        isr: {
+            // A secret random string that you create.
+            bypassToken: "005556d774a8",
+            // Paths that will always be served fresh.
+            exclude: [ "/api/invalidate" ] 
+        }
+    })
+})
+```
```

Comment:
``````suggestion

## Cache invalidation options

By default, ISR responses are cached for the duration of your deployment. You can further control caching by setting an `expiration` time or prevent caching entirely for certain routes.

### Time-based invalidation

You can change the length of time to cache routes this by configuring an `expiration` value in seconds:

```js
export default defineConfig({
    output: "server",
    adapter: vercel({
        isr: {
            // caches all pages on first request and saves for 1 day
            expiration: 60 * 60 * 24
        }
    })
})
```

### Manual invalidation

To implement Vercel's [Draft mode](https://vercel.com/docs/build-output-api/v3/features#draft-mode), or [On-Demand Incremental Static Regeneration (ISR)](https://vercel.com/docs/build-output-api/v3/features#on-demand-incremental-static-regeneration-isr), you can create a bypass token and provide it to the `isr` config along with the paths to exclude from caching: 

```js
export default defineConfig({
    output: "server",
    adapter: vercel({
        isr: {
            // A secret random string that you create.
            bypassToken: "005556d774a8",
            // Paths that will always be served fresh.
            exclude: [ "/api/invalidate" ] 
        }
    })
})
```
`````` 
I'm not sure we can just jump to random h5s out of the blue like that.  :smile: 

Some other subtle suggestions for tweaks:

- Create one header for these two options, that are both ways of controling/avoiding caches
- Put the default behaviour *there*, in the preamble, since it's appropriate for both (and good to state a default upfront)
- Be explicit about `expiration` being in seconds
- (gut feeling?) Since we are linking to *Vercel's* docs, and using their terms, call these things "Vercel's Draft mode.. etc." Especially since this is the Vercel adapter, I think it's perfectly fine to take some of the accountability off of us, and make it clear "Our Vercel adapter can handle these Vercel features" and make it clear this is not an "Astro" thing, just a compatibility with/implementation of Vercel thing.